app/src/main/java/org/dhis2/data/forms/dataentry/fields/FieldViewModel.java

```diff
@@ -43,7 +43,7 @@
     public abstract FieldViewModel withWarning(@NonNull String warning);
 
     @NonNull
-    public abstract FieldViewModel withError(@NonNull String error);
+    public abstract FieldViewModel withError(@Nullable String error);
```

Comment:
do you think it make sense having a method called withError with nullable parameter?.
Maybe we can check when are we calling it with null , maybe we can avoid `ScanView` to call `"withError"` with null value.
What do you think?

Reply:
Its better that way, I'm changing it.